telemetry/vscode/src/generate.ts

```diff
@@ -132,7 +132,10 @@ const commonMetadata = [
   'requestId',
   'requestServiceType',
   'result',
-]
+] as const
+
+/** Allowed to be manually specified in .record() calls */
+const allowedMetadata: typeof commonMetadata[number][] = ['awsRegion']
```

Comment:
why not do this for all `commonMetadata`? what is the difference between commonMetadata vs allowedMetadata ?

Reply:
ok, would `optionalMetadata` be a better name for this?

And a docstring on `commonMetadata`:

    /**
     * Fields set automatically by the telemetry client (thus application code
     * normally shouldn't set these because the value will be overridden).
     */